### DISCUSSION TITLE
Handle multiple servers with same prisma instance

### DISCUSSION THREAD

**Discussion by perroudsky** [prisma/prisma#19684] [Q&A] [answered]
Hi !
I'd like to create a second server that would handle cron jobs and long running tasks such as checking and modifying hundreds of rows. 

Given that he second server would introduce new models to the schema.prisma file, how can we achieve this without messing the prisma instance of the first server ?

I've seen here that we can create a monorepo but ideally i would not want to change the architecture of the main server.

Is this possible ?

Visual explanation of the use case:
```
mainClient ------- | mainServer | ---------|
                                            |
                                            --------- DATABASE
                                            |
backOffice ------- | secondServer | -------|   
```                                                               

Thanks :)

**Comment by nurul3101** [comment]
Hello @perroudsky 👋 

Based on your use case, You can have two schema files that point to the same database. And these schemas would contain the models relevant to that server. The key point is that all the models in both servers should exist in the database.

Here's an example reference on how you could achieve it:

- https://github.com/prisma/prisma/issues/2443#issuecomment-630679118

I would recommend that you should ensure that the schema is synchronized across all the servers, i.e., if you add a new field to model A in the main server, you should also add it to model A in the second server, and vice versa.

Hope this helps! Please feel free to let us know in case you have any additional queries.

**Reply by perroudsky** [reply]
Hello @nurul3101 ,
thanks for your answer!

I think the github issue you have linked is not related to my use case since I only need 1 database. (I’d like two servers pointing at the same database).

I understand that it’s not possible to have independent servers with their own models.
I assume it would cause problems for migrations ?

The only solution is to have synchronized schema across all the servers right?
But can you help me understand how can I do that ?

Thanks again for your help :)

**Reply by nurul3101** [reply, marked as answer]
You should have a single source of truth. You can create your own custom npm package that has the schema file, you can use this schema file from the custom npm package in both servers to keep it synchronised.

**Reply by nurul3101** [reply]
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
